Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -6372,3 +6399,623 @@ void CClientPed::UpdateStreamPosition(const CVector& vecInPosition)
     }
     CClientStreamElement::UpdateStreamPosition(vecPosition);
 }
+
+//////////////////////////////////////////////////////////////////
+//
+// CClientPed::EnterVehicle
+//
+// Asks server for permission to start entering vehicle
+//
+//////////////////////////////////////////////////////////////////
+bool CClientPed::EnterVehicle(CClientVehicle* pVehicle, bool bPassenger)
```

Comment:
- [ ] @Zangomangu  I think there should be a common prelude, like a private `ProcessInitialVehicleInOutKey` which booth calls and does these checks. Also note that for some reason these are actually currently inconsistent as well. 
![image](https://user-images.githubusercontent.com/923242/98430723-6c16e880-20a7-11eb-9f90-7a85c166840d.png)

Reply:
A common prelude would basically be: EnterOrExitVehicle(bool bEnterOrExit, bool bPassenger) to save a few of these initial checks.
It could be changed but is unnecessary imo.

The methods are split up and called EnterVehicle and ExitVehicle because they are now also called from scripting,

I added the check for dead ped. There is no reason to continue on if the ped is dead.

Reply:
> I added the check for dead ped.

Do you mean the check for the dead vehicle?

Reply:
Is there a particular reason checks are done in a different order for exits vs enters?

Reply:
The check for dead vehicle is not possible in EnterVehicle because we dont have a valid vehicle yet at that point.
As for the order of the checks, you want me to change that to match?